lib/auth/methods.go

```diff
@@ -540,6 +538,14 @@ func (a *Server) authenticateHeadless(ctx context.Context, req authclient.Authen
 		return nil, trace.AccessDenied("headless authentication public key mismatch")
 	}
 
+	// Only grab user lock on successful headless auth to perform normal login
+	// checks, like whether the user is locked. Failed headless login attempts
+	// won't lock out the user.
+	if err := a.WithUserLock(ctx, req.Username, func() error { return nil }); err != nil {
+		log.Debugf("WithUserLock for user %q failed during headless authentication: %v", req.Username, err)
+		return nil, trace.Wrap(authenticateHeadlessError)
+	}
```

Comment:
It means checking the user's failed login attempts to check whether the user is locked by failed logins, and updating the user's login attempts upon auth failure.

We actually don't need this for headless auth as it's more similar to SSO login than local login - it usually fails from timeout / cancellation. Even if the user denies a headless request we don't want to lock out the user, so it can just be removed. Updated the comment as well.